utils/json_test.go

```diff
@@ -0,0 +1,30 @@
+package utils
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestAndValidateCIDGeneration(t *testing.T) {
+	testCases := []struct {
+		name    string
+		content string
+	}{
+		{
+			"empty string", "",
+		},
+		{
+			"empty json", "{}",
+		},
+
+		{
+			"test record", "\\xa6curlohttps://cerc.iodtypex\\x19WebsiteRegistrationRecordgversione0.0.1ltls_cert_cidx.QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnRrbuild_artifact_cidx.QmP8jTG1m9GSDJLCbeWhVSVgEzCPPwXRdCRuJtQ5Tz9Kc9x\\x1crepo_registration_record_cidx.QmSnuWmxptJZdLJpKRarxBMS2Ju2oANVrgbr2xWbie9b2D",
+		},
+	}
+
+	for _, tc := range testCases {
+		deprecatedAndCorrect, _ := CIDFromJSONBytes([]byte(tc.content))
```

Comment:
We should either catch any errors or make sure the returned result isn't `nil`/empty, because if both functions were to error out and return a `nil`/empty result it would evaluate them as equal and the test would pass.